### PR TITLE
docs(typings): #6549 improve typings using overloads - getLocation, getSize

### PR DESCRIPTION
**Pull request by iamkenos**
## Proposed changes

- function overloads as suggested [here](https://github.com/webdriverio/webdriverio/issues/6549#issue-822350507)

the idea is to have something like this as a result:
Screenshot 2021-03-06 at 11 14 45 PM

[//]: # (Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.)

## Types of changes

[//]: # (What types of changes does your code introduce to WebdriverIO?)
[//]: # (_Put an `x` in the boxes that apply_)

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation update (typings)

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [x] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/main/CONTRIBUTING.md) doc
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] I have added proper type definitions for new commands (if appropriate)

## Further comments

1. The [issue](https://github.com/webdriverio/webdriverio/issues/6549) we're aiming to fix gave `getLocation` and `getSize` as examples. I skimmed other browser and element commands under the webdriverio package that may need the same change and couldn't find others. I could be mistaken though so would appreciate if someone can check that.

2. ~~Not really sure if I would need to add tests for this either, for now I did without.~~

3. There's a completely unrelated [commit](https://github.com/webdriverio/webdriverio/pull/6558/commits/413b4e2465752f01df182a7126995ac83a05faed) in this PR; i had to add this in because of [this recent change](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/51520/files) that broke the tests 

[//]: # (If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...)

### CODE REVIEW

File: packages/webdriverio/src/commands/element/getLocation.ts [hunk 1]
```diff
@@ -27,16 +36,11 @@ import { getElementRect } from '../../utils'
  * @uses protocol/elementIdLocation
  * @type property
  */
-export default async function getLocation (
+export async function getLocation<T extends keyof Coordinates> (
```
Comment (erwinheitzman) [inline packages/webdriverio/src/commands/element/getLocation.ts:39]: why export twice this method twice?

**Review by erwinheitzman — COMMENTED** [review]
Thank you for the contribution, love the idea. Left a quick question regarding the double export

**Comment by erwinheitzman** [comment]
We do have tests for our types and we would love to see these changes covered as well =]

File: packages/webdriverio/src/commands/element/getLocation.ts (hunk 1 above)
Comment (iamkenos) [reply packages/webdriverio/src/commands/element/getLocation.ts:39]: Hi! By twice, were you referring to the default export below or the named exports above?

**Comment by iamkenos** [comment]
> We do have tests for our types and we would love to see these changes covered as well =]

Apologies and thanks in advance for the patience, I may need some hand holding here 😅

Would you be able to point out a file I can refer to for some samples on typings ut please? Can't promise I can do it by i'll give it a try!

**Comment by erwinheitzman** [comment]
No worries, here you go: https://github.com/webdriverio/webdriverio/tree/main/tests/typings

**Comment by iamkenos** [comment]
> No worries, here you go: https://github.com/webdriverio/webdriverio/tree/main/tests/typings

thanks! added the tests too

**Comment by iamkenos** [comment]
> Thank you for the contribution, love the idea. Left a quick question regarding the double export

Removed redundant exports, thanks for highlighting! I ended up removing the generics, was getting the some weird issues with the return types and it turns out it was because of the default export.

**Review by christian-bromann — APPROVED** [review]
Great patch, thanks! 👍

**Pull request merged**